builder/cloudstack/step_create_instance.go

```diff
@@ -30,13 +30,14 @@ func (s *stepCreateInstance) Run(state multistep.StateBag) multistep.StepAction
 	client := state.Get("client").(*cloudstack.CloudStackClient)
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
+	source := state.Get("source").(string)
 
 	ui.Say("Creating instance...")
 
 	// Create a new parameter struct.
 	p := client.VirtualMachine.NewDeployVirtualMachineParams(
 		config.ServiceOffering,
-		config.instanceSource,
+		source,
```

Comment:
I would just use `state.Get("source").(string)` here, as this is the only place where we use `source`.